### PR TITLE
Toolbar layout fixes

### PR DESCRIPTION
**Pull request by chaosphere2112**
Fixes a number of minor layout/interaction issues with the toolbar, and adds tests that cover some of the basics (as well as some of the specific issues I encountered).

Ctests are passing for me, but I'm waiting on the upload (my home laptop does not love concurrent testing, so it'll be a bit).

Matching baselines coming shortly.

### CODE REVIEW

**Comment by chaosphere2112** [comment]
https://open.cdash.org/viewTest.php?onlypassed&buildid=3800440

**Comment by chaosphere2112** [comment]
@aashish24 @doutriaux1 @dlonie Here are the changes I mentioned in the Syncup email.

File: Packages/vcs/Lib/vtk_ui/button.py [hunk 1]
```diff
@@ -135,12 +135,20 @@ def add_state(self, label=None, image=None, bgcolor=None, fgcolor=None, opacity=
...
         self.repr.SetPlaceFactor(1)
         self.repr.PlaceWidget(bounds)
+        self.repr.Modified()
+        if self.showing():
+            # This One Weird Hack will make your Buttons Go In the Right Place - Developers hate it!
+            # Buttons weren't always getting properly placed (toolbars in toolbars being the canonical example)
+            # This makes them show up correctly. Weird, but it works.
+            h_state = self.repr.GetHighlightState()
+            self.repr.Highlight((h_state + 1) % 3)
+            self.repr.Highlight(h_state)
```
Comment (dlonie) [inline Packages/vcs/Lib/vtk_ui/button.py]: Nice comment :P

Strange. Looking at the C++, all this does is trigger a HighlightEvent (which, according to `git grep` doesn't seem to be handled anywhere internally by VTK, nor by uvcdat/vistrails), and calls Modified(), which you do above. Strange.

**Comment by dlonie** [comment]
LGTM

File: Packages/vcs/Lib/vtk_ui/button.py (hunk 1 above)
Comment (chaosphere2112) [inline Packages/vcs/Lib/vtk_ui/button.py]: Yeah, you're telling me. I actually have a tendency to delve down into the C++ source when I find a weird hack like this, so I can ideally *not* have the weird hack anymore. This time I failed.

**Pull request merged**